src/modules/navigator/navigator_main.cpp

```diff
@@ -996,24 +990,17 @@ Navigator::get_cruising_throttle()
 }
 
 float
-Navigator::get_acceptance_radius(float mission_item_radius)
+Navigator::get_acceptance_radius()
 {
-	float radius = mission_item_radius;
-
-	// XXX only use navigation capabilities for now
-	// when in fixed wing mode
-	// this might need locking against a commanded transition
-	// so that a stale _vstatus doesn't trigger an accepted mission item.
-
-	const position_controller_status_s &pos_ctrl_status = _position_controller_status_sub.get();
+	if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
+		// return the value specified in the parameter NAV_ACC_RAD
+		return get_default_acceptance_radius();
 
-	if (_vstatus.vehicle_type != vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
```

Comment:
@sfuhrer Are you sure that removing this logic does not have any side effects?

Reply:
@RomanBapst I guess you mainly refer to  `(pos_ctrl_status.timestamp > _pos_sp_triplet.timestamp)`? Do you see a logical reason where this could actually be false?
At least in SITL it wasn't yet a problem, but this specific part wasn't yet flight tested.

Reply:
It might be safer if you flip it around so that for FW (and rover) you're using pos_ctrl_status.acceptance_radius, otherwise fallback to using the default acceptance radius.

I would also keep a minimum pos_ctrl_status validity check, maybe just pos_ctrl_status.timestamp != 0 and finite.

Reply:
@dagar how about it now? a7a5303bb6a93850c2c373575f5587e15c547dbb